doc/design/scheduled-tasks.md

```diff
@@ -46,9 +46,16 @@ def scheduled_task():
 The job would then immediately set `version.needs_recomputing = False` on all of the "dirty" Versions.
 If any simultaneous requests happen to modify the Version while it is being recomputed/validated, those changes may or may not happen in time to be represented.
 However, they are guaranteed to trigger another recomputation in 5 minutes, so the system will eventually regain consistency.
+If the job takes too long to complete and the next job kicks off while the first is still running, this will also prevent any versions from being recomputed in different jobs.
```

Comment:
sorry, I am not sure what is `this` which "will also prevent any versions from being recomputed in different jobs", could you please elaborate? e.g. if we started computation at 5 min mark, it takes 7 min (very unlikely, but let's think hypothetically), what would prevent the next job at 10 min mark?

Reply:
gotcha, got `this` -- thanks!